discoverecs.py

```diff
@@ -297,6 +298,9 @@ def get_infos(self):
         clusters_pages = self.ecs_client.get_paginator("list_clusters").paginate()
         for clusters in clusters_pages:
             for cluster_arn in clusters["clusterArns"]:
+                # Only register selected ARNs.
+                if cluster_arn not in self.cluster_arns:
```

Comment:
```suggestion
                if len(self.cluster_arns) > 0 and cluster_arn not in self.cluster_arns:
```

This should continue to work as without the argument if no cluster arns are specified

Reply:
I thought about this while writing that line. I figured, since it's an empty list by default, `cluster_arn` won't be in it. That's why I felt a length-check was redundant.

I'm happy to go with your suggestion regardless, if you prefer it.

Reply:
I just tested the current implementation locally and it doesn't pull any definitions when no cluster arns are specified, pretty sure the length check is required else we skip every cluster

Reply:
It's probably preferable to add a `list_clusters` method, which returns `self.cluster_arns` if it exists, otherwise iterates as the current behavior, and then call that in `get_infos`

i.e.

```python
    def list_clusters(self):
        if self.cluster_arns:
			return self.cluster_arns
        cluster_arns = []
        clusters_pages = self.ecs_client.get_paginator("list_clusters").paginate()
        for clusters in clusters_pages:
            for cluster_arn in clusters["clusterArns"]:
            	cluster_arns += cluster_arn
		return cluster_arns
```

Reply:
Huh, I see. Thanks for the suggestion. I really like that idea. Sadly, I'm having a hard time getting it to work. I'm currently working with this function:

```python
    def list_clusters(self):
        if len(self.cluster_arns) > 0:
            return self.cluster_arns
        cluster_arns = []
        clusters_pages = self.ecs_client.get_paginator("list_clusters").paginate()
        for clusters in clusters_pages:
            for cluster_arn in clusters["clusterArns"]:
                cluster_arns += cluster_arn

        print(cluster_arns)
        return cluster_arns
```

In the case where I don't specify ARNs explicitly, I expect `print(cluster_arns)` to print a list of the ARNs of my ECS clusters. However, it's a list of the individual characters of those ARNs. My attempts to resolve this were not successful 😞

This leads to failures down the line. I figured the problem needs to be solved in this function already.

Reply:
And just after writing this, I find a resource that shows `cluster_arns += [cluster_arn]` as a way to append a string to a list of strings. I assume this is the way to go?

Reply:
Ah sorry yes that's a mistake in my suggestion (I don't write python often either)

You can either do as you suggested: `cluster_arns += [cluster_arn]`, or `append(cluster_arns, cluster_arn)`, either are fine and should work.

Reply:
Okay, awesome. I did that in my latest commit 😊